tensorflow/core/kernels/data/experimental/group_by_reducer_dataset_op.cc

```diff
@@ -243,8 +243,11 @@ class GroupByReducerDatasetOp : public UnaryDatasetOpKernel {
             if (states_.find(key) == states_.end()) {
               // Run the init function to create the initial state.
               std::vector<Tensor> init_func_output;
+              RecordStop(ctx);
```

Comment:
We almost certainly do not want to require callers of `Run` to be aware of the need to wrap the invocation in `RecordStop` and `RecordStart`. Since we are passing `IteratorContext` and `Node` into the method, I can imagine this could be folded into the `Run` method.

Reply:
`RecordStop` and `RecordStart` are now called before and after RunSync dispatch.